src/servers/http_server_v2.cc

```diff
@@ -1812,12 +1813,13 @@ HTTPAPIServerV2::EVBufferToInput(
       const char* shm_region = nullptr;
       if (CheckSharedMemoryData(
               request_input, &shm_region, &offset, &byte_size)) {
-        if (request_input.FindMember("data") == request_input.MemberEnd()) {
-          return TRTSERVER_ErrorNew(
-              TRTSERVER_ERROR_INVALID_ARG,
-              "must not specify 'data' field in request input when using "
-              "shared memory");
-        }
+        // Uncomment after client fix (Don't send data in case of shared memory)
+        // if (request_input.FindMember("data") == request_input.MemberEnd()) {
+        //   return TRTSERVER_ErrorNew(
+        //       TRTSERVER_ERROR_INVALID_ARG,
+        //       "must not specify 'data' field in request input when using "
+        //       "shared memory");
```

Comment:
The conditional conflicts with the error message. If `Find()` returns `End()`, then "data" is not specified.

Reply:
This is to verify that the infer request header does not contain 'data' fields. 
The docs specify that in the case of shared memory: the “data” field of request_input must not be set and the returned $response_output must not define the “data” field.

Reply:
Check again when will the conditional be evaluated to true. Do you intend to return error in that case?

Reply:
Fixed